src/Compilers/Core/VBCSCompiler/BuildProtocol.cs

```diff
@@ -182,9 +182,9 @@ internal class BuildRequest
 
                 // Grab the MemoryStream and its internal buffer
                 // to prevent making another copy.
-                var stream = (MemoryStream)writer.BaseStream;
+                var stream = writer.BaseStream;
                 // Write the length of the request
-                int length = (int)stream.Length;
+                int length = checked((int)stream.Length);
```

Comment:
Ah, so the checked() is protecting against underflow (wrapping to a negative int).